### ISSUE TITLE
Fix formatting in main.asm

### ISSUE THREAD

**Issue by yenatch** [pret/pokecrystal#70]
The existing main.asm is hard to follow. Formatting conventions are inconsistent and content is disorganized.

Proposed:
- 68-character width limit
- phase out tab-formatting
- split off large chunks of related asm into files
- more straightforward comments
- discourage use of addresses where possible

Example bank:

``` asm
; ========================= B A N K  3 A =========================
;            Sound engine and music/sound effect pointers

SECTION "bank3A", DATA, BANK[$3A]

; The entire sound engine. Interfaces are in bank 0
    INCLUDE "audio/engine.asm"

; List of what music plays when a trainer notices you
    INCLUDE "audio/music_encounters.asm"

; Pointer table for all the music in the game
    INCLUDE "audio/music_pointers.asm"
; Empty song
    INCLUDE "audio/music/nothing.s"

; Pointer table for all the cries in the game
    INCLUDE "audio/cry_pointers.asm"

; Pointer table for all the sound effects in the game
    INCLUDE "audio/sfx_pointers.asm"

; The remainder of the bank is taken up by music
Music_Route36:           INCLUDE "audio/music/route36.s"
Music_RivalBattle:       INCLUDE "audio/music/rivalbattle.s"
Music_RocketBattle:      INCLUDE "audio/music/rocketbattle.s"
Music_ElmsLab:           INCLUDE "audio/music/elmslab.s"
Music_DarkCave:          INCLUDE "audio/music/darkcave.s"
Music_JohtoLeader:       INCLUDE "audio/music/johtoleader.s"
Music_Champion:          INCLUDE "audio/music/champion.s"
Music_SSAqua:            INCLUDE "audio/music/ssaqua.s"
Music_NewBarkTown:       INCLUDE "audio/music/newbarktown.s"
Music_GoldenrodCity:     INCLUDE "audio/music/goldenrodcity.s"
Music_VermilionCity:     INCLUDE "audio/music/vermilioncity.s"
Music_TitleScreen:       INCLUDE "audio/music/titlescreen.s"
Music_RuinsOfAlphInside: INCLUDE "audio/music/ruinsofalphinside.s"
Music_LookPokemaniac:    INCLUDE "audio/music/lookpokemaniac.s"
Music_TrainerVictory:    INCLUDE "audio/music/trainervictory.s"


; ========================= B A N K  3 B =========================
;                              Songs

SECTION "bank3B", DATA, BANK[$3B]

; ...
```

**Comment by Caverns4** [comment]
I would change:
[code]; ========================= B A N K  3 A =========================
[/code]

to something more like
[code]; === Bank  0x3A ===[/code]
But I do feel this is a good thing to do, if nothing else just make things easier to find.

In fact, I would personally like to see things even further subdivided where possible, and have other things merged into bank files. With how important ROM banks are with this processor, each bank might be best split into a file perhaps?

**Comment by kanzure** [comment]
Yep, `0x3a` is much better than `3 A`. When a reader searches a file, they are going to be looking for 0x3 or "bank3a" not "bank 3 a".

**Comment by Caverns4** [comment]
They also might be looking for Bank $3A, since the $ also denotes hex, but 0x3A would be perfectly suitable.
